WorkloadManagementSystem/Client/WMSClient.py

```diff
@@ -138,13 +143,49 @@ def submitJob( self, jdl, jobDescriptionObject = None ):
       return result
 
     # Submit the job now and get the new job ID
+    bulkTransaction = self.operationsHelper.getValue( 'JobScheduling/BulkSubmissionTransaction', False )
+    if bulkTransaction:
+      result = getParameterVectorLength( classAdJob )
+      if not result['OK']:
+        return result
+      nJobs = result['Value']
+      parametricJob = nJobs > 0
+
     if not self.jobManager:
       self.jobManager = RPCClient( 'WorkloadManagement/JobManager',
                                    useCertificates = self.useCertificates,
                                    timeout = self.timeout )
-    result = self.jobManager.submitJob( classAdJob.asJDL() )
-    if 'requireProxyUpload' in result and result['requireProxyUpload']:
+    if bulkTransaction and parametricJob:
+      gLogger.debug( 'Applying transactional job submission' )
+      resultSubmit = self.jobManager.submitJob( classAdJob.asJDL(), True )
+      if resultSubmit['OK'] and resultSubmit.get( 'requireBulkSubmissionConfirmation' ):
+        jobIDList = resultSubmit['Value']
+        if len( jobIDList ) == nJobs:
+          # Confirm the submitted jobs
+          confirmed = False
+          for attempt in range(3):
+            result = self.jobManager.confirmBulkSubmission( jobIDList )
+            if result['OK']:
+              confirmed = True
+              break
+            time.sleep( 1 )
+          if not confirmed:
+            # The bulk submission failed, try to delete the created jobs
+            result = self.jobManager.deleteJob( jobIDList )
+            error = "Job submission failed to confirm bulk transaction"
+            if not result['OK']:
+              error += "; removal of created jobs failed"
+            return S_ERROR( EWMSBULK, error )
+        else:
+          return S_ERROR( EWMSBULK, "The number of submitted jobs does not match job description ")
+      else:
+        result = resultSubmit
+    else:
+      resultSubmit = self.jobManager.submitJob( classAdJob.asJDL() )
+      result = resultSubmit
+    if resultSubmit.get( 'requireProxyUpload' ):
```

Comment:
Is it on purpose that you just print a warning and not actually do the upload ?

Reply:
This is on purpose. The proxy upload is done starting from the certificate (not proxy), using dirac-proxy-init or equivalent command